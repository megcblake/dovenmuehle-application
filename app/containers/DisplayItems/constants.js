export const LOAD_ITEMS = 'app/DisplayItems/LOAD_ITEMS';
export const LOAD_ITEMS_SUCCESS = 'app/DisplayItems/LOAD_ITEMS_SUCCESS';
export const LOAD_ITEMS_ERROR = 'app/DisplayItems/LOAD_ITEMS_ERROR';
