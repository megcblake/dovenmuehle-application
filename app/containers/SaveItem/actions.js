import {
  INPUT_ITEM,
  SAVE_ITEM,
  SAVE_ITEM_SUCCESS,
  SAVE_ITEM_ERROR,
} from './constants';

export function changeItem(item) {
  return {
    type: INPUT_ITEM,
    item,
  };
}

export function saveItem(item) {
  return {
    type: SAVE_ITEM,
    item,
  };
}

export function saveItemSuccess() {
  return {
    type: SAVE_ITEM_SUCCESS,
  };
}

export function saveItemError() {
  return {
    type: SAVE_ITEM_ERROR,
  };
}
