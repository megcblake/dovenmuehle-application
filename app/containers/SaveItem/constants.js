export const INPUT_ITEM = 'app/SaveItem/INPUT_ITEM';
export const SAVE_ITEM = 'app/SaveItem/SAVE_ITEM';
export const SAVE_ITEM_SUCCESS = 'app/SaveItem/SAVE_ITEM_SUCCESS';
export const SAVE_ITEM_ERROR = 'app/SaveItem/SAVE_ITEM_ERROR';
