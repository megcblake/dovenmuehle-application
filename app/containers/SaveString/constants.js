export const INPUT_STRING = 'dovenmuehele/Save/INPUT_STRING';
