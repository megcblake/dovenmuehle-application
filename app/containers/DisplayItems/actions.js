import { LOAD_ITEMS, LOAD_ITEMS_SUCCESS, LOAD_ITEMS_ERROR } from './constants';

export function loadItems() {
  return {
    type: LOAD_ITEMS,
  };
}

export function loadItemsSuccess(items) {
  return {
    type: LOAD_ITEMS_SUCCESS,
    items,
  };
}

export function loadItemsError(error) {
  return {
    type: LOAD_ITEMS_ERROR,
    error,
  };
}
