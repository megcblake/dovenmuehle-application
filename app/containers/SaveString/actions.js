import { INPUT_STRING } from './constants';

export function changeString(string) {
  return {
    type: INPUT_STRING,
    string,
  };
}
