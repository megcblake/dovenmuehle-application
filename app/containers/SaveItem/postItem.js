import { put, select } from 'redux-saga/effects';
import { saveItemSuccess, saveItemError } from './actions';
import { itemSelector } from './selectors';

export function* postItem() {
  const item = yield select(itemSelector());
  try {
    yield fetch('/api/saveItem', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ item }),
    });
    yield put(saveItemSuccess());
  } catch (error) {
    yield put(saveItemError());
  }
}
