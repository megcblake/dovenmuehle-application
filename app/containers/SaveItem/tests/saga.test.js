import { put, takeLatest } from 'redux-saga/effects';
import { saveItemError } from '../actions';
import { postItem } from '../postItem';
import { SAVE_ITEM } from '../constants';
import displayItemsSaga from '../saga';

describe('displayItemsSaga', () => {
  it('should call on SAVE_ITEM', () => {
    const iterator = displayItemsSaga();
    const expected = takeLatest(SAVE_ITEM, postItem);
    const actual = iterator.next().value;

    expect(actual).toEqual(expected);
  });
});

describe('postItem', () => {
  let saveItemGenerator;

  beforeEach(() => {
    saveItemGenerator = postItem();

    const selectDescriptor = saveItemGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should call the saveItemError action if the response errors', () => {
    const putDescriptor = saveItemGenerator.next().value;
    expect(putDescriptor).toEqual(put(saveItemError()));
  });
});
