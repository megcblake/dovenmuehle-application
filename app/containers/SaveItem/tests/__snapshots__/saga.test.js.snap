// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postItem should call the saveItemError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;
